Lesson11/Task2.py

```diff
@@ -0,0 +1,192 @@
+import json
+
+
+def load_phonebook():
+    try:
+        with open('phonebook.json') as file:
+            phonebook = json.load(file)
+    except FileNotFoundError:
+        phonebook = []
+
+    return phonebook
+
+
+def save_phonebook(phonebook):
+    with open('phonebook.json', 'w', encoding='utf-8') as file:
+        json.dump(phonebook, file, indent=4, ensure_ascii=False)
+
+
+def add_contact():
+    name = input("Введіть ім'я: ")
+    surname = input("Введіть фамілію: ")
+    full_name = f"{name} {surname}"
+    phone_number = input("Введіть номер телефону: ")
+    city = input("Введіть місто: ")
+    country = input("Введіть країну: ")
+
+    contact = {
+        'name': name,
+        'surname': surname,
+        'full_name': full_name,
+        'phone_number': phone_number,
+        'city': city,
+        'country': country
+    }
+
+    phonebook = load_phonebook()
+    phonebook.append(contact)
+    save_phonebook(phonebook)
+    print("Контакт успішно доданий!")
+
+
+def search_by_name(name):
+    phonebook = load_phonebook()
+    found_contacts = []
+    for contact in phonebook:
+        if contact['name'] == name:
+            found_contacts.append(contact)
+
+    return found_contacts
+
+
+def search_by_surname(surname):
+    phonebook = load_phonebook()
+    found_contacts = []
+    for contact in phonebook:
+        if contact['surname'] == surname:
+            found_contacts.append(contact)
+
+    return found_contacts
+
+
+def search_by_full_name(full_name):
+    phonebook = load_phonebook()
+    found_contacts = []
+    for contact in phonebook:
+        if contact['full_name'] == full_name:
+            found_contacts.append(contact)
+
+    return found_contacts
+
+
+def search_by_phone_number(phone_number):
+    phonebook = load_phonebook()
+    found_contacts = []
+    for contact in phonebook:
+        if contact['phone_number'] == phone_number:
+            found_contacts.append(contact)
+
+    return found_contacts
+
+
+def search_by_location(location):
+    phonebook = load_phonebook()
+    found_contacts = []
+    for contact in phonebook:
+        if contact['city'] == location or contact['country'] == location:
+            found_contacts.append(contact)
+
+    return found_contacts
+
+
+def delete_contact(phone_number):
+    phonebook = load_phonebook()
+    updated_phonebook = [contact for contact in phonebook if contact['phone_number'] != phone_number]
+    save_phonebook(updated_phonebook)
+    print("Контакт успішно видалений!")
+
+
+def update_contact(phone_number):
+    phonebook = load
+
+
+def update_contact(phone_number):
+    phonebook = load_phonebook()
+    for contact in phonebook:
```

Comment:
Two similar function names. Please cleanup

Reply:
Hi, @Masergio  could you check again, I've made some changes in the code, from 124 lines (Task2.py) and Task(1) - it's an old one.